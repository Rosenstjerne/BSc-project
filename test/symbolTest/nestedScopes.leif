var int a;

function int outer_function(int a) {
    var int b 
    b = a * 2;

    function int inner_function(int c) {
        return c * b;
    }

    return inner_function(3);
}

a = 5;

print(outer_function(a)); # Should print 30