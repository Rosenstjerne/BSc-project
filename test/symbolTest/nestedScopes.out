function null main() {
	var int a;
	
	function int outer_function(int a) {
		var int b;
		
		function int inner_function(int c) {
			return (c * b);
			
		}
		
		b = (a * 1);
		return inner_function(3);
		
	}
	
	a = 5;
	print(outer_function(a));
	
}
